warehouse/redshift.go

```diff
@@ -46,6 +47,37 @@ func NewRedshift(c *config.Config) *Redshift {
 	}
 }
 
+func (rs *Redshift) getExportTableName(fullyQualifiedName bool) string {
+	if rs.conf.Redshift.DatabaseSchema == "search_path" || !fullyQualifiedName {
+		return rs.conf.Redshift.ExportTable
+	}
+	return fmt.Sprintf("%s.%s", rs.conf.Redshift.DatabaseSchema, rs.conf.Redshift.ExportTable)
+}
+
+func (rs *Redshift) getSyncTableName(fullyQualifiedName bool) string {
+	if rs.conf.Redshift.DatabaseSchema == "search_path" || !fullyQualifiedName {
+		return rs.conf.Redshift.SyncTable
+	}
+	return fmt.Sprintf("%s.%s", rs.conf.Redshift.DatabaseSchema, rs.conf.Redshift.SyncTable)
+}
+
+func (rs *Redshift) getSchemaParameter() string {
+	// the built-in current_schema() function will walk the Postgres search_path to get a schema name
+	// more info: https://www.postgresql.org/docs/9.4/functions-info.html
+	if rs.conf.Redshift.DatabaseSchema == "search_path" {
+		return "current_schema()"
+	}
+
+	return fmt.Sprintf("'%s'", rs.conf.Redshift.DatabaseSchema)
+}
+
+func (rs *Redshift) validateSchemaConfig() error {
+	if rs.conf.Redshift.DatabaseSchema == "" {
```

Comment:
was wondering if no schema is defined, then relying on current schema is a good default? From what I've understood about the hauser user base, a majority are not power users, and probably just use the default public schema.

Reply:
That was my original thought as well, but I came to a different conclusion after further consideration. The behavior of hauser when the DatabaseSchema parameter is empty would not have been clear and could have resulted in putting data in an unintended location. By not accepting a blank parameter and returning a link to documentation when it occurs, we can prevent this type of confusion.

To be sure, using the public schema may not be an issue for some businesses, but our enterprise customers have lately expressed a need to have more control over where the export data gets migrated to in their data warehouse.

Reply:
I should also mention: I updated example-config.toml with `DatabaseSchema = "public"`

New users will likely start with this file when creating their own configs, so they would expect the public schema to be used unless the intentionally changed the config to use a different schema.

Reply:
SGTM!